ci-operator/step-registry/ipi/conf/ibmcloud/manual-creds/ipi-conf-ibmcloud-manual-creds-commands.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+TARGET_DIR="${SHARED_DIR}"
+PROVIDER=ibmcloud
+
+oc adm release extract --credentials-requests=true --cloud="${PROVIDER}" --to="${TARGET_DIR}" "${RELEASE_IMAGE_LATEST}"
```

Comment:
What is this about? Also, notice that this will run on the CI cluster.

Reply:
We are forced to generate credentials manifests since we have to use `Manual` credentials mode.
The alternative is attempting to manually create credentials, like AWS does with OIDC, but we should now have support to allow the `release` command to generate them automatically for us. Without these credentials, I don't think the cluster will work (the operators won't be able to authenticate I believe).

We need the manifests created prior to creating the test cluster, so I had thought this would be the best time to run the command and store those manifests in the shared directory so they can be copied over prior to creating the cluster.

Reply:
How often those manifests are changed? Why can't we have them in a secret?

Reply:
I'm not sure on that actually, let me ask dev on that

Reply:
Because we have an automated way to generate them in every 12hours by typing that command.

Reply:
So the manifests will be a `CredentialsRequest` resource YAML definition. Which looks like they will access the necessary credentials via a `secretRef` in that YAML defintion. Some examples of these manifests that are used, based on cloud provider. For instance, here's the change to add IBM Cloud support for `machine-api-operator`
https://github.com/openshift/machine-api-operator/pull/871/files#diff-88dafc65725c017015e420841dbf7ba59aadcfc05420b9d38d92bf8becd31dfd

So, the credentials themselves are stored in secrets in the IPI deployed cluster, and these generated manifests of `CredentialsRequest` for each supported operator, will handle the necessary requests. This way, we don't have to manually generate credentials and can use these requests instead.

The number of generated manifests would potentially change, as we add support for more operators in the future, but I don't think the defined manifests themselves change too often.

So if we would rather not use `oc adm release extract`, we could instead maintain the set of manifests manually and update the `ipi-conf-ibmcloud-manual-creds-commands.sh` whenever we add more in the future. But the contents of the manifests generated appear to all be stored in Github in individual repos. At least that's how I understand all of it, the `ClusterRequest` manifest for each operator is taken from those available, such as from this list in `machine-api-operator`
https://github.com/openshift/machine-api-operator/blob/master/install/0000_30_machine-api-operator_00_credentials-request.yaml